enhancements/cluster-logging/forwarder-to-s3.md

```diff
@@ -0,0 +1,224 @@
+* Re-use existing AWS authentication features provided by the Cloudwatch output
+
+### Non-Goals
+
+* To provide an API the exposes all the configuration points of the underlying collector implementation
+
+## Proposal
+
+This enhancement proposes to:
+
+* Enhance the ClusterLogForwarder API to add an S3 output
+  * Define a default schema for writing log records to an S3 bucket that is based
+upon the log type and source which is consistent with other output types
+  * Allow the schema for writting log records to be modified by the administrator
+  * Reuse the authorization mechanisms that are available with the Cloudwatch output
+* Add a generator to support generating collector configuration based upon the spec defined in the ClusterLogForwarder API
+
+
+### Workflow Description
+
+**cluster administrator** is a human responsible for administering the **cluster-logging-operator**
+and **ClusterLogForwarders**
+
+1. The cluster administrator deployes the cluster-logging-operator if it is already not deployed
+1. The cluster administrator edits or creates a **ClusterLogForwarder** and defines an S3 output
+1. The cluster administrator references the S3 output in a pipeline
+1. The cluster-logging-operator reconciles the **ClusterLogForwarder**, generates a new collector configuration,
+and updates the collector deployment
+
+### API Extensions
+
+#### ClusterLogForwarder API
+
+```yaml
+apiVersion: "observability.openshift.io/v1"
+kind: ClusterLogForwarder
+spec:
+outputs:
+- name:
+  type:                    # add s3 to the enum
+  tls: {}
+  s3:
+    url:                   # (optional)
+    region:                # (optional) string that is different from the configured service default
+    bucker:                # string for the S3 bucket absent leading 's3://' or trailing '/'
+    keyPrefix:             # (optional) templated string (see note 1)
+    authentication:
+      type:                # enum: awsAccessKey, iamRole
+      awsAccessKey:
+        assumeRole:        # (optional)
+        roleARN:           # secret reference
+        externalID:        # (optional) secret reference
+      iamRole:
+        roleARN:           # secret reference
+        token:             # bearer token
+        assumeRole:        # (optional)
+          roleARN:         # secret reference
+          externalID:      # (optional) secret reference
+    tuning:
+      delivery:            # (optional) atLeastOnce, atMostOnce
+      maxWrite:            # (optional) quantity (e.g. 500k)
+      compression:         # (optional) none, gzip,zstd,snappy,zlib
+      minRetryDuration:    # (optional) duration
+      maxRetryDuration:    # (optional) duration
+```
+
+**Note 1:** A combination of static and dynamic values consisting of field paths followed by "\|\|" followed by another field path or a static value (e.g `foo.{.bar.baz\|\|.qux.quux.corge\|\|.grault\|\|"nil"}-waldo.fred{.plugh\|\|"none"}`)
```

Comment:
to we have to explicitly call out datetime formatting here?

I had this dir tree in mind for our use-case:

Directory Tree:
    625f0e4d-1f8a-4434-8de3-02ded5acdfde/  				# Cluster ID
        2025-08-13/  								# Date
            1755102268-28e968da-cac7-41a5-8fbf-4357625ca81a.log.gz gz  	# Log File
            1755102570-41ac683c-83a3-465a-9e85-d39aa1a742ec.log.gz	# Log File
            1755102815-1d17fd1a-0f4d-4801-a742-48991fb366e6.log.gz gz	# Log File
            1755103241-aa894078-c2bc-4261-9461-0126ba18ff0d.log.gz	# Log File

Reply:
I have concerns about how one might accomplish injecting datetime formatting into these template fields because of it opening the door to add more functions.  We intentionally do not add this to the API because we wish to avoid the complexities that vector currently has with VRL injection into the configuration.

That being said, it looks like the [default](https://vector.dev/docs/reference/configuration/sinks/aws_s3/#filename_time_format) settings are to pre-append the date to the key prefix which need to be accounted for in the implementation in order to accomplish what you request.  Maybe @alanconway has some ideas